grammar.js

```diff
@@ -1191,6 +1197,7 @@ module.exports = grammar(C, {
     _assignment_left_expression: ($, original) => choice(
       original,
       $.qualified_identifier,
+      $.user_defined_literal,
```

Comment:
Can you show me an example where this is needed/used? I don't think this is generally valid?

UDLs return an instance of a class or some primitive, and it's generally not legal to assign to prvalues (the type of the UDL expression). It's basically like writing:

```cpp
// "hello"_str = "world";
std::string("hello") = "world";
```

Reply:
First case of https://github.com/tree-sitter/tree-sitter-cpp/issues/220